### ISSUE TITLE
Specifying the `training` flag in GANs causes mode collapse

### ISSUE THREAD

**Issue by SirDavidLudwig** [DLii-Research/deep-learning-dna#4]
The implementation of the AC-GAN, perhaps even the vanilla GAN, suffers from mode collapse that does not occur with a manual custom training loop.

**Comment by SirDavidLudwig** [comment]
Removing the training flag from the `train_step` method resolves the mode collapse issue, but this is not an acceptable solution...

**Comment by SirDavidLudwig** [comment]
The issue is temporarily resolved with an improper solution.

**Comment by SirDavidLudwig** [comment]
https://github.com/tensorflow/addons/issues/2414

**Comment by SirDavidLudwig** [comment]
Custom spectral norm implementations still seem to result in mode collapse... Maybe it's integrated into the models incorrectly? Aside from the mode collapse, it trains OK...

EDIT: Or maybe not...

**Comment by SirDavidLudwig** [comment]
![image](https://user-images.githubusercontent.com/13006946/175765376-3908d79f-7d3e-49ac-96ce-8768a831442a.png)
 8 Epochs

**Comment by SirDavidLudwig** [comment]
Closing for now with the conclusion that spectral normalization is just bad for these types of training regimes. The original paper used the Geometric GAN training regime, which may have been more helpful.
